src/components/TextInput.tsx

```diff
@@ -0,0 +1,37 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core'
+import React, { DetailedHTMLProps, InputHTMLAttributes } from 'react'
+import { CommonStyleProps, useTheme } from '../style'
+
+type HTMLInputProps = DetailedHTMLProps<
+  InputHTMLAttributes<HTMLInputElement>,
+  HTMLInputElement
+>
+
+type Props = HTMLInputProps &
+  CommonStyleProps & {
+    label?: string
+    type?:
+      | 'date'
+      | 'email'
+      | 'month'
+      | 'number'
+      | 'password'
+      | 'tel'
+      | 'text'
+      | 'time'
+      | 'url'
+  }
+
+export const TextInput: React.FC<Props> = (props) => {
```

Comment:
It can be used for numbers, but also we can omit the label if we want. Maybe this should be only `Input`, and then create Components for all the other `input.type`, e.g. `Radio` (for HTML `<input type='radio'/>`), `Checkbox`, etc.